Create rule S6562: Always set the DateTimeKind when creating a new DateTime object (#1712)
SonarSource/rspec: rules/S6562/impact.adoc

> Creating the `DatTime` object without specifying the kind explicitly will set the default value of `DateTimeKind.Unspecified`. In this case calling the method `ToUniversalTime` will assume it was a local time while calling `ToLocalTime` will assume the time is UTC. This might lead to mismatched `DateTime` objects in your application.

Reviewer: ```suggestion
Creating the `DateTime` object without specifying the property `Kind` will set it to the default value of `DateTimeKind.Unspecified`. In this case, calling the method `ToUniversalTime` will assume that `Kind` is `DateTimeKind.Local` and calling the method `ToLocalTime` will assume that it's `DateTimeKind.Utc`.
As a result, you might have mismatched `DateTime` objects in your application.
```
